apps/experiments/management/commands/duplicate_shared_pipelines.py

```diff
@@ -0,0 +1,91 @@
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import Count
+
+from apps.experiments.models import Experiment
+from apps.pipelines.models import Node, Pipeline
+
+
+class Command(BaseCommand):
+    help = "Find pipelines that are shared by multiple experiments and create individual copies\
+    for each experiment (except one)"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Show what would be done without making changes",
+        )
+
+    def handle(self, *args, **options):
+        dry_run = options["dry_run"]
+
+        shared_pipelines = Pipeline.objects.annotate(experiment_count=Count("experiment")).filter(
```

Comment:
I think this should also filter by `working_version=None` since we only want to duplicate working version pipelines right?

Reply:
I think so yeah. [Pipelines are archived with experiments](https://github.com/dimagi/open-chat-studio/blob/c4651c54c11137fe4ccd14634a175e3a231e4193/apps/experiments/models.py#L911-L928), but in the case of versions, there will always be a 1-1 relationship between the pipeline and experiment, since we create a version of the pipeline when versioning. Only working experiments have 1-N rel to pipelines currently.

Reply:
right thank you both